x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/use_create_analytics_form.ts

```diff
@@ -125,25 +125,68 @@ export const useCreateAnalyticsForm = (): CreateAnalyticsFormProps => {
 
   const createKibanaIndexPattern = async () => {
     const dataViewName = destinationIndex;
+    let retryCount = 15;
 
     try {
-      await mlContext.dataViewsContract.createAndSave(
-        {
-          title: dataViewName,
-        },
-        false,
-        true
-      );
+      const interval = setInterval(async () => {
```

Comment:
From my experience intervals can be dangerous as they can easily be broken cause infinite loops. Having multiple paths and `clearInterval` calls can be hard to follow for a maintainer and they could easily break it.

If I were writing this I would use a timeout inside a function that can then calls itself again, only if the index doesn't exist and the `retryCount` is greater than 0.

Reply:
Replaced interval in 5c5f24f7f64192367d2665034a6d914076cbd6cf